lib/tag_parser.rb

```diff
@@ -0,0 +1,54 @@
+
+module TagParser
+
+
+  def self.parse_tag(string)
+
+    type = string.match /<(\w+)/
+    type = type[1] if type
```

Comment:
Could probably just do type = string.match(/regex/)[1]
If there isn't a match for some reason what happens (type = nil)? Checking "if type" is fine but right here doesn't seem to do much since it goes forward either way.